modules/gltf/src/gltf/gltf-post-processor.js

```diff
@@ -0,0 +1,310 @@
+import {getBytesFromComponentType, getSizeFromAccessorType} from '../utils/gltf-type-utils';
+
+const GL_SAMPLER = {
+  TEXTURE_MAG_FILTER: 0x2800,
+  TEXTURE_MIN_FILTER: 0x2801,
+  TEXTURE_WRAP_S: 0x2802,
+  TEXTURE_WRAP_T: 0x2803
+};
+
+const SAMPLER_PARAMETER_GLTF_TO_GL = {
+  magFilter: GL_SAMPLER.TEXTURE_MAG_FILTER,
+  minFilter: GL_SAMPLER.TEXTURE_MIN_FILTER,
+  wrapS: GL_SAMPLER.TEXTURE_WRAP_S,
+  wrapT: GL_SAMPLER.TEXTURE_WRAP_T
+};
+
+export default class GLTFPostProcessor {
+
+  postProcess(gltf, options = {}) {
+    this.gltf = gltf;
+    this._resolveToTree(options);
+    return this.gltf;
+  }
+
+  _postProcessGLTF(options = {}) {
+    // Create all images (if requested)
+    this.out.images = (this.gltf.images || [])
+      .map(image => this.parseImage(image, options))
+      .filter(Boolean);
+
+    // Normalize all scenes
+    this.out.scenes = (this.gltf.scenes || [])
+      .map(scene => this.parseScene(scene, options))
+      .filter(Boolean);
+
+    if (this.gltf.scene !== undefined) {
+      this.out.scene = this.gltf.scenes[this.gltf.scene];
+    }
+
+    return this;
+  }
+
+  // Convert indexed glTF structure into tree structure
+  // PREPARATION STEP: CROSS-LINK INDEX RESOLUTION, ENUM LOOKUP, CONVENIENCE CALCULATIONS
+  /* eslint-disable complexity */
+  _resolveToTree(options = {}) {
```

Comment:
Should we also resolve **animations**?

Reply:
> Should we also resolve animations?

Yes that would seem to make sense. I will leave that up to you since you are doing the work on animations.